sphinx/source/docs/dev_guide/documentation.rst

```diff
@@ -175,71 +198,145 @@ When building Bokeh's documentation, the most common options for ``make`` are:
+  modules and their properties. These texts are available from the Python
+  interpreter and within most Python development environments. Sphinx also uses
+  those texts to generate the :ref:`reference guide <refguide>` section of
+  Bokeh's documentation.
 
 In the file :bokeh-tree:`sphinx/source/rst_epilog.txt`, you can find many common
 substitutions used across the narrative documentation as well as docstrings and
 model help texts. This file is loaded as the
 `epilog.rst environment variable`_ for Bokeh's Sphinx configuration.
 
+.. _devguide_documentation_edit_narrative:
+
+Writing for Bokeh's narrative documentation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Bokeh's narrative documentation consists of these four elements:
+
+* :ref:`first_steps`: first steps guides and installation instructions
+* :ref:`userguide`: descriptions and instructions for using Bokeh
+* :ref:`gallery`: interactive examples with source code
+* :ref:`Contributors guide <contributors_guide>`: instructions for contributing to Bokeh
+
+Sphinx generates each of those elements from reStructuredText (.rst) files. To
+edit any of those elements, open the corresponding ReST source file in the
+``sphinx/source/docs`` folder of the Bokeh source tree.
+
+For information on how to format text using reStructuredText, see the
+`reStructuredText primer on the Sphinx website`_ or the `official
+reStructuredText website`_.
+
+For information on writing style, see Bokeh's
+:ref:`documentation style guidelines <devguide_documentation_style_guidelines>`
+and the `Google developer documentation style guide`_.
+
+.. _devguide_documentation_edit_source:
+
 Contributing to Bokeh's source code documentation
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-All functions and methods in Bokeh use docstrings_. In addition, Bokeh uses its
-own system to provide `detailed information on individual properties`_.
+All functions and methods in Bokeh use
+:ref:`docstrings <devguide_documentation_edit_docstrings>`. In addition, Bokeh
+uses its own system to provide
+:ref:`detailed information on individual properties <devguide_documentation_edit_properties_help>`.
 
-.. _docstrings:
+.. _devguide_documentation_edit_docstrings:
 
 Writing docstrings
 ''''''''''''''''''
 
-To automatically process all docstrings, Bokeh uses an extension for Sphinx
+To automatically process Python docstrings, Bokeh uses an extension for Sphinx
 called `Napoleon`_ with `Napoleon's Google style`_. For Napoleon to work
-correctly, all docstrings you write should follow the rules in the `Google
+correctly, all docstrings you write need to follow the rules in the `Google
 Python Style Guide`_.
 
 Docstrings for functions and methods generally include these three elements:
 
-* A short description of what the function or method does, starting with a
+* A short description of what the function does, starting with a
   verb. For example: "Creates and returns a new Foo."
```

Comment:
```suggestion
  verb. For example: "Create and return a new Foo."
```
would be consistent with the voice I've tried to use throughout since the beginning